rasa/core/policies/flow_policy.py

```diff
@@ -614,94 +559,97 @@ def _run_step(
         A tuple of the predicted action and a list of events.
         """
         if isinstance(step, QuestionFlowStep):
-            structlogger.debug("flow.step.run.question", step=step, flow=flow)
-            slot = tracker.slots.get(step.question, None)
-            initial_value = slot.initial_value if slot else None
-            slot_value = slot.value if slot else None
-            if step.skip_if_filled and slot_value != initial_value:
-                return ActionPrediction(None, 0.0)
+            structlogger.debug("flow.step.run.question")
+
+            context = self.flow_stack.current_context().copy()
+            context["question"] = step.question
+
+            self.flow_stack.push(
+                FlowStackFrame(
+                    flow_id=FLOW_PATTERN_ASK_QUESTION,
+                    frame_type=StackFrameType.REMARK,
+                    context=context,
+                )
+            )
 
-            question_action = ActionPrediction("question_" + step.question, 1.0)
-            if slot_value != initial_value:
-                question_action.events = [SlotSet(step.question, initial_value)]
-            return question_action
+            return ActionPrediction(None, 0.0)
```

Comment:
More for my knowledge, an action prediction with these parameters allows the assistant to continue making predictions? 🤔

Reply:
yes indeed, but improved the code to make that more explicit.